source/volume_of_fluid/reconstruct.cc

```diff
@@ -0,0 +1,488 @@
+/*
+ Copyright (C) 2016 - 2018 by the authors of the ASPECT code.
+
+ This file is part of ASPECT.
+
+ ASPECT is free software; you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation; either version 2, or (at your option)
+ any later version.
+
+ ASPECT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#include <aspect/global.h>
+#include <aspect/simulator.h>
+#include <aspect/volume_of_fluid/handler.h>
+#include <aspect/volume_of_fluid/utilities.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  template <>
+  void VolumeOfFluidHandler<2>::update_volume_of_fluid_normals (const VolumeOfFluidField<2> field,
+                                                                LinearAlgebra::BlockVector &solution)
+  {
+    const int dim = 2;
+    const unsigned int max_degree = 1;
+
+    LinearAlgebra::BlockVector initial_solution;
+
+    sim.computing_timer.enter_section("   Reconstruct VolumeOfFluid interfaces");
+
+    initial_solution.reinit(sim.system_rhs, false);
+
+    // Boundary reference
+    typename DoFHandler<dim>::active_cell_iterator endc =
+      this->get_dof_handler().end ();
+
+    // Interface Reconstruction vars
+    const unsigned int n_local = 9;
+
+    Vector<double> local_volume_of_fluids (n_local);
+    std::vector<Point<dim>> resc_cell_centers (n_local);
+    std::vector<typename DoFHandler<dim>::active_cell_iterator> neighbor_cells(n_local);
+
+    const unsigned int n_sums = 3;
```

Comment:
Same here, what type of sums? Remember this code will be read by other people a lot of times, every minute we spend now on making it more readable will save others hours of work.